announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -2806,6 +2944,15 @@ else if (announceTo != null && announceTo.equals("groups"))
 						// set access
 						header.setGroupAccess(groups);
 					}
+					if (announceTo != null && announceTo.equals("roles"))
+					{
+						msg.getPropertiesEdit().addProperty("selectedRoles", String.valueOf(state.getTempAnnounceToRoles()));
+						header.clearGroupAccess();
```

Comment:
clear group access?

Reply:
Yes, It is used it when an announcement it changes to groups view into a roles view and delete all the groups in the header